lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -648,6 +650,17 @@ defmodule Kernel.ErrorsTest do
     assert_compile_fail CompileError,
       "nofile:1: cannot use ^x outside of match clauses",
       'x = 8; <<a, b::size(^x)>> = <<?a, ?b>>'
+
+    warning = capture_io(:stderr, fn ->
```

Comment:
Let's move the test to the kernel/expansion_test. It is already async: false and arguably a better place for this test since it does happen during expansion.